db/version_builder.cc

```diff
@@ -81,8 +81,11 @@ class VersionBuilder::Rep {
   };
 
   struct LevelState {
-    std::unordered_set<uint64_t> deleted_files;
-    // Map from file number to file meta data.
+    // Files in base version that should be deleted.
+    std::unordered_set<uint64_t> deleted_base_files;
+    // Files moved from base version.
+    std::unordered_map<uint64_t, FileMetaData*> moved_files;
```

Comment:
There is one difference, only `added_files` will be deref-ed when destructing this `VersionBuilder`. It is done so to avoid manipulating `moved_files` reference counter without DB mutex.

Reply:
yeah, I later realize the difference.